### PR TITLE
Feature/melos

### PR DESCRIPTION
**Pull request by Leptopoda**
superseeds #333 
fixes: #326 
first step towards: #177

We could also do something like:

```yaml
scripts:
  format: dart format --fix --line-length 120 .
  gen-l10n:
    exec: flutter gen-l10n && melos run format
    packageFilters:
      flutter: true
      fileExists: l10n.yaml
````

So we wouldn't need our custom tools. But there isn't any benefit to it either.

### CODE REVIEW

**Comment by Leptopoda** [comment]
Ok I killed CI. Melos reccomends not adding the dependency_overrides to the version control but in our case this might be favorable as we don't use melos in CI.

**Comment by Leptopoda** [comment]
The error could be related to
https://github.com/firebase/flutterfire/blob/a55ce608c374de17ddbb7cd86b7bf47d4730b381/melos.yaml#L26-L30

**Comment by Leptopoda** [comment]
and we still have some not overridden deps
https://github.com/provokateurin/nextcloud-neon/pull/336/files#diff-880fef6d3249a857ddf338262e812c10a96265ac30dd77e443c3649cfa1a2f9bR516-R527

File: pubspec.yaml
[the remark this answers is not in the document]
Comment (provokateurin) [reply pubspec.yaml:1]: I think it's ok.

**Comment by provokateurin** [comment]
Maybe it's easier to just completely switch to melos for everything.

**Comment by Leptopoda** [comment]
> Maybe it's easier to just completely switch to melos for everything.

long term probably (that's why I marked the melos issue as `first step` and not as `implements:`) but I don't think it's on scope for this PR.

I think I can fix the current issues to finally have a working nextcloud lib again and personally don't really enjoy configuring CI so I'd leave that up to someone else (maybe you :P)

**Comment by provokateurin** [comment]
Ok that's fair! Let's fix the imports in this PR and I'll migrate CI later :)

**Review by provokateurin — APPROVED** [review]

**Pull request merged**